onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -259,8 +259,8 @@ common::Status NnapiExecutionProvider::Compile(const std::vector<FusedNodeAndGra
       builder.SetTargetDeviceOption(nnapi::ModelBuilder::TargetDeviceOption::CPU_ONLY);
     }
 
-    std::unique_ptr<nnapi::Model> nnapi_model;
-    ORT_RETURN_IF_ERROR(builder.Compile(nnapi_model));
+    ORT_RETURN_IF_ERROR(builder.Compile());
+    std::unique_ptr<nnapi::Model> nnapi_model(&builder.GetModel());
 
```

Comment:
I think the original setup was fine here. The ModelBuilder internally creates the nnapi::Model and owns it until Compile is called, at which point the Compile implementation std::move's it into the arg.

Right now you have ModelBuilder owning the model, and when Compile is called you store a pointer to it in a unique_ptr meaning that unique_ptr will now delete the model, but so will the ModelBuilder dtor. 

The value of a unique_ptr should only ever be set via a std::move or an allocation.